pkg/scheduler/scheduler.go

```diff
@@ -410,10 +414,17 @@ func (e entryOrdering) Less(i, j int) bool {
 	return a.Obj.CreationTimestamp.Before(&b.Obj.CreationTimestamp)
 }
 
-func (s *Scheduler) requeueAndUpdate(log logr.Logger, ctx context.Context, w *workload.Info, message string) {
-	added := s.queues.RequeueWorkload(ctx, w)
-	log.V(2).Info("Workload re-queued", "queuedWorkload", klog.KObj(w.Obj), "queue", klog.KRef(w.Obj.Namespace, w.Obj.Spec.QueueName), "added", added)
-	err := workload.UpdateWorkloadStatus(ctx, s.client, w.Obj, kueue.QueuedWorkloadAdmitted, corev1.ConditionFalse, "Pending", message)
+func (s *Scheduler) requeueAndUpdate(log logr.Logger, ctx context.Context, e entry) {
+	if e.skipped {
+		if !s.queues.AddOrUpdateWorkload(e.Obj) {
+			log.V(2).Info("Queue for workload didn't exist; ignored for now", "queuedWorkload", klog.KObj(e.Obj), "queue", klog.KRef(e.Obj.Namespace, e.Obj.Spec.QueueName))
+		}
+		return
+	}
+
+	added := s.queues.RequeueWorkload(ctx, &e.Info)
+	log.V(2).Info("Workload re-queued", "queuedWorkload", klog.KObj(e.Obj), "queue", klog.KRef(e.Obj.Namespace, e.Obj.Spec.QueueName), "added", added)
+	err := workload.UpdateWorkloadStatus(ctx, s.client, e.Obj, kueue.QueuedWorkloadAdmitted, corev1.ConditionFalse, "Pending", e.noAdmissionReason)
```

Comment:
I think we should not update the workload status if the workload was skipped;

Reply:
it returns in line 422